controllers/tenant_controller.go

```diff
@@ -79,6 +80,22 @@ func (r TenantReconciler) Reconcile(request ctrl.Request) (result ctrl.Result, e
 		return reconcile.Result{}, err
 	}
 
+	// Validate ingressClasses regexp
+	if len(instance.Spec.IngressClasses.AllowedRegex) > 0 {
+		if _, err := regexp.Compile(instance.Spec.IngressClasses.AllowedRegex); err != nil {
+			r.Log.Info("Unable to compile allowedRegex for ingressClasses", "allowedRegex", instance.Spec.IngressClasses.AllowedRegex)
+			return reconcile.Result{}, nil
+		}
+	}
+
+	// Validate storageClasses regexp
+	if len(instance.Spec.StorageClasses.AllowedRegex) > 0 {
+		if _, err := regexp.Compile(instance.Spec.StorageClasses.AllowedRegex); err != nil {
```

Comment:
Probably we should move these `regexp.Compile` at a Tenant validation webhook, so the Cluster Admin can easily check if there's a typo in those expressions.

Right now, it's ok here: just let's plan it as Issue/Enhancement.

Reply:
Let's do it right now and rename `tenant_owner` webhook to `tenant`